File Mover/fmover.py

```diff
@@ -0,0 +1,248 @@
+                            if operation == "Copy":
+                                result = shutil.copy(get_path('src') + "/" + file, sc.sortbytype(file_ending) + "/" + file)
+                            else:
+                                result = shutil.move(get_path('src') + "/" + file, sc.sortbytype(file_ending) + "/" + file)
+                        elif is_file_in_dst_dir is False and get_subdir() is True:
+                            result = None
+                            if operation == 'Copy':
+                                result = shutil.copy(get_path('src') + "/" + file, sc.sortbytype(file_ending) + "/" + file)
+                            else:
+                                result = shutil.move(get_path('src') + "/" + file, sc.sortbytype(file_ending) + "/" + file)
+                        else:
+                            pass
+
+            return sg.PopupOK(f"File transfer successful!\nFile(s) moved to '{get_path('dst')}'")
+
+
+fmover = FileMover()
+
+source = get_path('src')
+destination = get_path('dst')
+
+
+def get_subdir():
+    if os.path.exists(str(destination) + '/' + 'Images'):
+        return True
+
+    elif os.path.exists(str(destination) + '/' + 'Archives'):
+        return True
+
+    elif os.path.exists(str(destination) + '/' + 'Text Files'):
+        return True
+
+    else:
+        return False
+
+
+
+class SortCriteria():
+
+    def sortbytype(self, ftype):
+        type_list = [ftype]
+        image_list = ['.png', '.jpg', '.jpeg', '.gif']
+        archive_list = ['.zip', '.rar', '.7z']
+        textf_list = ['.txt', '.md']
+
+        # For image files
+        for type in type_list:
+            if type in image_list: #'.png' or '.jpg' or '.jpeg' or '.gif':
+                if os.path.exists(str(destination) + '/' + 'Images'):
+                    return str(os.path.join(str(destination) + '/' + 'Images'))
+                else:
+                    os.mkdir(str(destination) + '/' + 'Images')
+                    return str(os.path.join(str(destination) + '/' + 'Images'))
+
+        # For archive files
+            elif type in archive_list:
+                if os.path.exists(str(destination) + '/' + 'Archives'):
+                    return str(os.path.join(str(destination) + '/' + 'Archives'))
+                else:
+                    os.mkdir(str(destination) + '/' + 'Archives')
+                    return str(os.path.join(str(destination) + '/' + 'Archives'))
+
+        # For text files
+            elif type in textf_list:
+                    if os.path.exists(str(destination) + '/' + 'Text Files'):
+                        return str(os.path.join(str(destination) + '/' + 'Text Files'))
+                    else:
+                        os.mkdir(str(destination) + '/' + 'Text Files')
+                        return str(os.path.join(str(destination) + '/' + 'Text Files'))
+
+            else:
+                sg.PopupError("File type not found!")
+                raise SystemExit()
+
```

Comment:
Delete these empty lines

Reply:
@avinashkranjan I've deleted them. Somehow I missed that xD